### DISCUSSION TITLE
Contributor Motivation: Gamification with Achievements

### DISCUSSION THREAD

**Discussion by MartinThoma** [py-pdf/pypdf#2185] [Ideas]
I am thinking about how we can motivate new contributors and how we can keep existing contributors motivated.

The basis is certainly a friendly, appreciating atmosphere + as little "bureaucracy" as possible (randomly failing CI, strictly following processes e.g. how a new feature should be introduced).

Besides that, I was thinking about adding an "achievement" system:

* Bulldozer: Found 10+ bugs
* Bug Hunter: Closed 10+ bugs (@pubpub-zz you gave me the idea xD)
* Cartographer: Labeled 10+ issues

We could make a community page with who got those achievements, similar to https://stackoverflow.com/help/badges/4369/refiner

**Comment by stefan6419846** [comment]
Speaking for myself, I do not consider gamification a required part for FOSS contributions in general and for PyPDF as one of the FOSS projects and it probably would not be the reason for me keeping to contribute to PyPDF. If I find something where I can contribute to and have enough time while seeing a value in the change, I will add a corresponding contribution. This is how I experienced most other FOSS projects as well.

The specific issue I see with PyPDF is that it requires a rather deep understanding of the PDF specification, which generally is not really widespread, and there are lots of tools which introduce all sorts of standard violations/edge cases as well. While I personally work a lot with PDFs, I did not really have the time and need to study the PDF specs in detail for now, as there are much more areas of interest and work to consider which have a higher priority for me.

By the way: The proposed *Cartographer* level cannot be achieved by non-members of the repository, as besides adding the labels from the issue template, users cannot manage the issue tags as far as I am aware.

**Reply by MartinThoma** [reply]
Regarding the labels: I'm also happy when people point out in the comments which labels I should add (mainly for old issues; helping people to find them + reminding me that they exist)

**Reply by MartinThoma** [reply]
But I also hope that we get some time to 0 open issues. Let's see :smile:

**Comment by stefan6419846** [comment]
This has no gotten much traction besides my comment and with me being the current de-facto maintainer and more or less not seeing a future with this approach, I am going to close this for now.
